### PR TITLE
feat: update indexing store with new checkpoint encoding

### PR DESCRIPTION
**Pull request by jaylmiller**

### CODE REVIEW

File: packages/core/src/database/postgres/service.test.ts [hunk 1]
```diff
@@ -63,15 +63,19 @@ describe.skipIf(shouldSkip)("postgres database", () => {
...
-      "kysely_migration",
-      "kysely_migration_lock",
-      "function_metadata",
-      "table_metadata",
-      "instance_metadata",
-      "0xPet",
-      "0xPerson",
-    ]);
+    expect(
+      new Set(await getTableNames(database.db, "ponder_cache")),
```
Comment (0xOlias) [inline packages/core/src/database/postgres/service.test.ts:67]: Could also do `.sort()` on both arrays here. Unfortunately, don't think vitest has a matcher for arrays ignoring order.

File: packages/core/src/database/postgres/service.ts [hunk 2]
```diff
@@ -244,9 +247,9 @@ export class PostgresDatabaseService implements BaseDatabaseService {
         functionId: m.function_id,
         functionName: m.function_name,
         fromCheckpoint: m.from_checkpoint
-          ? decodeCheckpoint(m.from_checkpoint)
+          ? decodeCheckpoint(m.from_checkpoint.toString())
           : null,
-        toCheckpoint: decodeCheckpoint(m.to_checkpoint),
+        toCheckpoint: decodeCheckpoint(m.to_checkpoint.toString()),
```
Comment (0xOlias) [inline packages/core/src/database/postgres/service.ts:252]: Are these still necessary now that we're using VARCHAR?

File: packages/core/src/database/postgres/service.ts [hunk 3]
```diff
@@ -377,34 +380,35 @@ export class PostgresDatabaseService implements BaseDatabaseService {
...
+                  tx,
+                ),
               );
 
               // Truncate cache tables to match metadata.
               await tx
                 .withSchema(CACHE_SCHEMA_NAME)
                 .updateTable(tableId)
-                .set({ effective_to: "latest" })
+                .set({ effective_to: encodeCheckpoint(maxCheckpoint) })
```
Comment (0xOlias) [inline packages/core/src/database/postgres/service.ts:392]: Could use the `LATEST` const from above here

File: packages/core/src/indexing-store/postgres/store.ts [hunk 4]
```diff
@@ -23,6 +27,7 @@ import {
 const MAX_BATCH_SIZE = 1_000 as const;
 const DEFAULT_LIMIT = 50 as const;
 const MAX_LIMIT = 1_000 as const;
+const LATEST = encodeCheckpoint(maxCheckpoint);
```
Comment (0xOlias) [inline packages/core/src/indexing-store/postgres/store.ts:30]: Might just make sense to export `LATEST` from `utils/checkpoint.js` seeing how much it's used

File: packages/core/src/database/postgres/service.ts (hunk 2 above)
Comment (jaylmiller) [reply packages/core/src/database/postgres/service.ts:252]: good call forgot to remove these!

File: packages/core/src/sync-store/store.ts [hunk 5]
```diff
@@ -181,6 +181,7 @@ export interface SyncStore {
         log: Log;
         block: Block;
         transaction: Transaction;
+        checkpoint?: Checkpoint;
```
Comment (kyscott18) [inline packages/core/src/sync-store/store.ts:184]: When would checkpoint be undefined?

File: packages/core/src/database/postgres/migrations.ts [hunk 6]
```diff
@@ -35,6 +35,24 @@ const migrations: Record<string, Migration> = {
         .execute();
     },
   },
+  "2024_03_22_new_checkpoint_encoding": {
+    async up(db: Kysely<any>) {
+      await db.schema
+        .alterTable("function_metadata")
+        .alterColumn("from_checkpoint", (cb) => cb.setDataType("varchar(75)"))
```
[the remark this answers is not in the document]
Comment (jaylmiller) [reply packages/core/src/database/postgres/migrations.ts:42]: if it's numeric and the checkpoint has leading zeros they will be stripped

[the remark this answers is not in the document]
Comment (jaylmiller) [reply packages/core/src/database/postgres/migrations.ts:42]: theres no way to recover the actual checkpoint after leading zeros are stripped

File: packages/core/src/bin/utils/run.ts [hunk 7]
```diff
@@ -130,7 +129,7 @@ export async function run({
     functionIds,
   });
 
-  indexingService.processEvents();
+  indexingService?.processEvents();
```
[the remark this answers is not in the document]
Comment (jaylmiller) [reply packages/core/src/bin/utils/run.ts:132]: no i forgot to remove that while debugging

File: packages/core/src/sync-store/store.ts (hunk 5 above)
Comment (jaylmiller) [reply packages/core/src/sync-store/store.ts:184]: the log's tables checkpoint column is currently nullable. but i think once it gets to this part of the code its impossible to be undefined so i can change it

Comment (jaylmiller) [reply packages/core/src/sync-store/store.ts]: fixed :)

**Comment by 0xOlias** [comment]
Merged into #770

**Pull request closed without merging**